lint

```diff
@@ -172,6 +174,34 @@ lint_files() {
     exit $lint_result
 }
 
+matches_any() {
+    local filename="$1"
+    local patterns="$2"
+    IFS=$'\n' # Make newlines the only separator.
+    set -f    # Disable globbing so $patterns can be treated as is.
+    for pattern in $patterns; do
```

Comment:
Important: You don't appear to be resetting `IFS` after changing it here (or re-enabling globbing), which will cause major weirdness elsewhere.
Not as important: I'd prefer the `while read` pattern here instead of manipulating IFS and global flags:
```
echo "$patterns" | while read -r pattern; do
    ...
done
```
or
```
while read -r pattern; do
    ...
done <<<"$patterns"